<launch>

    <arg name="solver" default="OpenSot"/>
    <arg name="prefix" default=""/>
    <arg name="use_xbot_config" default="false"/>
    <arg name="verbosity" default="2"/>
    <arg name="rate" default="100.0"/>
    <arg name="tf_prefix" default="ci"/>
    <arg name="markers" default="true"/>
    <arg name="namespace" default=""/> <!-- dummy argument avoids pass_all_args error in parent launch file -->
    <arg name="robot" default=""/>
       
    
    <node pkg="cartesian_interface" type="ros_server_node" 
                                    name="ros_server_node" 
                                    required="true" 
                                    output="screen" 
                                    launch-prefix="$(arg prefix)">
                                    
        <param name="is_model_floating_base" type="bool" value="true"/>
        <param name="model_type" value="RBDL"/>
        <param name="solver" value="$(arg solver)"/>
        <param name="use_xbot_config" value="$(arg use_xbot_config)"/>
        <param name="verbosity" value="$(arg verbosity)"/>
        <param name="rate" value="$(arg rate)"/>
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
        
    </node>

    <node if="$(arg markers)" pkg="cartesian_interface" type="marker_spawner" name="interactive_markers" output="screen">
        <param name="tf_prefix" value="$(arg tf_prefix)"/>
    </node>

    

</launch>
